packages/beacon-node/src/chain/shufflingCache.ts

```diff
@@ -178,14 +178,19 @@ export class ShufflingCache implements IShufflingCache {
     this.insertPromise(epoch, decisionRoot);
     /**
      * TODO: (@matthewkeil) This will get replaced by a proper build queue and a worker to do calculations
-     * on a NICE thread with a rust implementation
+     * on a NICE thread
      */
-    callInNextEventLoop(() => {
-      const timer = this.metrics?.shufflingCache.shufflingCalculationTime.startTimer({source: "build"});
-      const shuffling = computeEpochShuffling(state, activeIndices, epoch);
-      timer?.();
-      this.set(shuffling, decisionRoot);
-    });
+    const timer = this.metrics?.shufflingCache.shufflingCalculationTime.startTimer({source: "build"});
+    computeEpochShufflingAsync(state, activeIndices, epoch)
+      .then((shuffling) => {
+        this.set(shuffling, decisionRoot);
+      })
+      .catch((err) =>
+        this.logger?.error(`error building shuffling for epoch ${epoch} at decisionRoot ${decisionRoot}`, err)
```

Comment:
2nd param is context, not the err

```suggestion
        this.logger?.error(`error building shuffling for epoch ${epoch} at decisionRoot ${decisionRoot}`, {}, err)
```

Reply:
Good catch!  Updated.